File: PRACTICA/Ruby/Irrgarten/fuzzy_player.rb
require_relative 'player'
require_relative 'dice'

class Fuzzy_player < Player
  def initialize(other)
    super(other)
  end

  def move(direction, valid_moves)
    size = valid_moves.size()
    contained = valid_moves.include?(direction)

    if((size > 0) && !contained)
      return Dice.next_step(direction, valid_moves, @intelligence)
    else
      return direction
    end
  end

  def attack()
    return sum_weapons + Dice.intensity(@strength)
  end

  def defensive_energy()
    return sum_shields + Dice.intensity(@intelligence)
  end

  def to_s
    return "Fuzzy " + super.to_s
  end

  public_class_method :new
end

File: PRACTICA/Java/irrgarten/FuzzyPlayer.java
package irrgarten;

import java.util.ArrayList;

public class FuzzyPlayer extends Player {
    public FuzzyPlayer(Player other) {
        super(other);
    }

    @Override
    public Directions move(Directions direction, ArrayList<Directions> validMoves) {
        return Dice.nextStep(direction, validMoves, getIntelligence());
    }

    @Override
    public float attack() {
        return sumWeapons() + Dice.intensity(getStrength());
    }

    @Override
    public float defensiveEnergy() {
        return sumShields() + Dice.intensity(getIntelligence());
    }

    @Override
    public String toString() {
        return "Fuzzy " + super.toString();
    }
}
